spec/features/devise/confirmations_new_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.feature 'Resend confirmation instructions', type: :feature do
+  scenario 'User requests resend with unconfirmed email' do
+    # Create a user with unconfirmed email
+    create(:user, email: 'user@example.com', password: 'password', confirmed_at: nil)
+
+    visit new_user_confirmation_path
+
+    within('.signup-container') do
+      expect(page).to have_field('Email', type: 'email')
+      expect(page).to have_selector('input[type="submit"].signup-submit', visible: :hidden)
+    end
+
+    fill_in 'Email', with: 'user@example.com'
+    execute_script("document.getElementById('confirmation-form').submit();")
```

Comment:
-  [ ] The tests are still failing as seen below. please fix them:


<img width="623" alt="image" src="https://github.com/ometman/budget-app/assets/52934128/b43d2000-85ac-41e8-8f9a-3e6cb0bc8555">

Reply:
All tests passes. Some will fail if the server is on during the test, in the Capybara environment.